pkg/operator/controller/ingress_status.go

```diff
@@ -0,0 +1,130 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// syncIngressStatus computes the current status of ic and therefrom
+// creates or updates the IngressController resource.
+func (r *reconciler) syncIngressStatus(ic *operatorv1.IngressController) error {
+	deployment, err := r.getIngressState(ic)
+	if err != nil {
+		return fmt.Errorf("failed to get IngressController state: %v", err)
+	}
+
+	oldStatus := ic.Status.DeepCopy()
+	ic.Status.Conditions = computeIngressStatusConditions(oldStatus.Conditions, deployment)
+
+	if !ingressStatusesEqual(*oldStatus, ic.Status) {
+		err = r.Client.Status().Update(context.TODO(), ic)
+		if err != nil {
+			return fmt.Errorf("failed to update IngressController %s/%s", ic.Namespace, ic.Name)
+		}
+	}
+	return nil
+}
+
+// getIngressState gets and returns the resources necessary to compute the
+// ingress controller's current state.
+func (r *reconciler) getIngressState(ic *operatorv1.IngressController) (*appsv1.Deployment, error) {
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: ic.Namespace, Name: ic.Name}, ic)
```

Comment:
I thought it would be a good idea to make the call just in case something happened to the ic during reconciliation. Probably overkill, so I'll remove.